### PR TITLE
Skip Integration tests on Public runners if not full tests needed

### PR DESCRIPTION
**Pull request by potiuk**
Sometimes integration tests fail with not being able to allocate ports in Public GitHub runners. It seems to be more problematic in Postgres/MySQL/MsSQL jobs where resource requirements are far bigger than in SQLite jobs. And in most PRs success in SQLite means that the tests will also pass on other backends.

We still have the safety net of "Canary" builds where complete set of tests is run so this should be safe to skip integration tests where public runners are infolved on non-sqlite backends.

This should significantly decrease number of false-negatives that non-commiters get in their PRs.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by Taragolis** [comment]
Finger crossed

File: dev/breeze/src/airflow_breeze/commands/testing_commands.py [hunk 1]
```diff
@@ -285,6 +286,13 @@ def run_tests_in_parallel(
                     if test_type.startswith(heavy_test_type):
                         test_types_list.remove(test_type)
                         tests_to_run_sequentially.append(test_type)
+            if full_tests_needed:
```
Comment (jedcunningham) [inline dev/breeze/src/airflow_breeze/commands/testing_commands.py]: Should this be `if not full_tests_needed`? We want to still run them when `full_tests_needed` is true, right?

Comment (potiuk) [reply dev/breeze/src/airflow_breeze/commands/testing_commands.py]: Obviously :facepalm: . That would actually explain why they were run last time :)

File: dev/breeze/src/airflow_breeze/utils/selective_checks.py [hunk 2]
```diff
@@ -290,7 +290,7 @@ def default_constraints_branch(self) -> str:
         return self._default_constraints_branch
 
     @cached_property
-    def _full_tests_needed(self) -> bool:
```
Comment (ferruzzi) [inline dev/breeze/src/airflow_breeze/utils/selective_checks.py:293]: Congrats, little method, you've been promoted.  :P

**Review by ferruzzi — APPROVED** [review]
Looks like a clean fix.  Let's hope it fixes the issue.  :+1: .

**Review by jedcunningham — APPROVED** [review]

**Comment by potiuk** [comment]
Actually I found that out that it did not work for Postgres - because it was inside "mysql', 'mssql' - moved the if outside

**Comment by potiuk** [comment]
And added better messaging (colour and showing up the actually skipped/sequentialized test types.

**Review by ephraimbuddy — APPROVED** [review]

**Comment by potiuk** [comment]
Failures unrelated. All good.

**Pull request merged**